src/config.rs

```diff
@@ -0,0 +1,327 @@
+use pyo3::prelude::*;
+use pyo3::types::PyType;
+use pyo3::{exceptions::PyValueError, types::PyDict, FromPyObject, IntoPy, PyAny, PyObject, PyResult, Python};
+
+#[derive(Debug, Clone, Default)]
+pub struct CoreConfig {
+    pub title: Option<String>,
+    pub strict: Option<bool>,
+    pub extra_fields_behavior: Option<ExtraBehavior>,
+    pub typed_dict_total: Option<bool>,
+    pub from_attributes: Option<bool>,
+    pub loc_by_alias: Option<bool>,
+    pub revalidate_instances: Option<RevalidateInstances>,
+    pub validate_default: Option<bool>,
+    pub populate_by_name: Option<bool>,
+    pub str_max_length: Option<i32>,
+    pub str_min_length: Option<i32>,
+    pub str_strip_whitespace: Option<bool>,
+    pub str_to_lower: Option<bool>,
+    pub str_to_upper: Option<bool>,
+    pub allow_inf_nan: Option<bool>,
+    pub ser_json_timedelta: Option<SerJsonTimedelta>,
+    pub ser_json_bytes: Option<SerJsonBytes>,
+    pub ser_json_inf_nan: Option<SerJsonInfNan>,
+    pub hide_input_in_errors: Option<bool>,
+    pub validation_error_cause: Option<bool>,
+    pub coerce_numbers_to_str: Option<bool>,
+    pub regex_engine: Option<RegexEngine>,
+    pub cache_strings: Option<CacheStrings>,
+}
+
+impl TryFrom<Bound<'_, PyDict>> for CoreConfig {
+    type Error = PyErr;
+    fn try_from(value: Bound<'_, PyDict>) -> Result<Self, Self::Error> {
+        Ok(CoreConfig {
+            title: value.get_item("title")?.map(|v| v.extract()).transpose()?,
```

Comment:
Maybe we use a helper function to reduce the repetition with the `?.map(|v| v.extract()).transpose()?`?

Reply:
Maybe a helper function and a macro would be a good combo here